src/systems/battery_plugin/LinearBatteryPlugin.cc

```diff
@@ -439,10 +480,26 @@ void LinearBatteryPlugin::PreUpdate(
 {
   IGN_PROFILE("LinearBatteryPlugin::PreUpdate");
 
-  // \todo(anyone) Add in the ability to stop the battery from draining
-  // after it has been started by a topic. See this comment:
-  // https://github.com/ignitionrobotics/ign-gazebo/pull/1255#discussion_r770223092
-  this->dataPtr->startDraining = this->dataPtr->startDrainingFromTopics;
+  // Recaculate the toal power load among consumers
+  double total_power_load = this->dataPtr->initialPowerLoad;
+  _ecm.Each<components::BatteryPowerLoad>(
+    [&](const Entity & /*_entity*/,
+        const components::BatteryPowerLoad *_batteryPowerLoadInfo)->bool
+    {
+      if (_batteryPowerLoadInfo->Data().batteryName ==
```

Comment:
Meta: Rather than checking battery names we could check if they belong to the same parent model?

Reply:
Does that make them unique? You cannot have two battering belonging to the same parent model?

Reply:
As per VC:  For simplicity we check for entities, the mapping name <==> entity can be done on the side of whoever is setting the battery consumer.